src/app/auth/register/register.component.html

```diff
@@ -0,0 +1,85 @@
+        <i class="material-icons">{{ registerForm.controls['password'].hasError('required') ||
+        registerForm.controls['password'].hasError('hasCapitalCase') ? 'cancel' :
+          'check_circle' }}</i>
+        Must contain at least 1 in Capital Case!
+      </label>
+      <label class="col"
+             [ngClass]="registerForm.controls['password'].hasError('required') || registerForm.controls['password'].hasError('hasSmallCase')  ? 'text-danger' : 'text-success'">
+        <i class="material-icons">{{ registerForm.controls['password'].hasError('required') ||
+        registerForm.controls['password'].hasError('hasSmallCase') ? 'cancel' :
+          'check_circle' }}</i>
+        Must contain at least 1 Letter in Small Case!
+      </label>
+      <label class="col"
+             [ngClass]="registerForm.controls['password'].hasError('required') || registerForm.controls['password'].hasError('hasSpecialCharacters') ? 'text-danger' : 'text-success'">
+        <i class="material-icons">{{ registerForm.controls['password'].hasError('required') ||
+        registerForm.controls['password'].hasError('hasSpecialCharacters') ? 'cancel' :
+          'check_circle' }}</i>
+        Must contain at least 1 Special Character!
+      </label>
+    </div>
+    <!--    <div *ngIf="registerForm.controls['password'].touched-->
+    <!--                                     && registerForm.controls['password'].hasError('pattern')">-->
+    <!--      <app-password-check [password]="registerForm.controls['password'].value"></app-password-check>-->
+    <!--    </div>-->
+    <mat-form-field appearance="fill">
+      <mat-label>confirm Password</mat-label>
+      <input matInput type="password" name="confirm-password-input" formControlName="confirmPassword"
+             [readonly]="!registerForm.controls['password'].value">
+      <mat-error *ngIf="!registerForm.controls['password'].value">
+        please fill field password first
+      </mat-error>
+      <!--      <mat-error *ngIf="registerForm.controls['confirmPassword'].touched-->
+      <!--                                          && registerForm.controls['confirmPassword'].hasError('required')">-->
+      <!--        password is required-->
+      <!--      </mat-error>-->
+      <mat-error *ngIf="registerForm.controls['confirmPassword'].touched
+                                          && !registerForm.controls['confirmPassword'].hasError('required')
+                                          && (registerForm.controls['password'].value
+                                     != registerForm.controls['confirmPassword'].value)">
+        confirm password not match
+      </mat-error>
+    </mat-form-field>
+    <button [disabled]="!registerForm.valid" (click)="submit()"
+            [ngStyle]="{'background-color': !registerForm.valid ? 'gray' :'green', 'cursor' : !registerForm.valid ?'':'pointer'} ">
```

Comment:
move conditional styling to css using [:disabled selector](https://developer.mozilla.org/en-US/docs/Web/CSS/:disabled)